airflow/models/dag.py

```diff
@@ -1690,6 +1702,9 @@ def set_task_instance_state(
             include_upstream=False,
         )
 
+        if dag_run_id:
+            execution_date = dag_run.execution_date
```

Comment:
Should this be merged into the `if dag_run_id` block above?

Reply:
It will cause us to have both an` execution_date` and `dag_run_id` sent to `set_state` which accepts exactly one value

Reply:
I kind of feel bad we need to rely on passing `None` (instead of not passing) for these arguments, but I guess it’s easiest. Maybe we should give these `execution_date`s different names instead of reusing the variable name? `input_execution_date` (which can be None) and `resolved_execution_date` (which is always datetime), for example.